src/plugins/discover/public/application/components/context_app/context_app.test.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { waitFor } from '@testing-library/react';
+import { mountWithIntl } from '@kbn/test/jest';
+import { createFilterManagerMock } from '../../../../../data/public/query/filter_manager/filter_manager.mock';
+import { mockTopNavMenu } from './__mocks__/top_nav_menu';
+import { ContextAppContent } from './context_app_content';
+import { indexPatternMock } from '../../../__mocks__/index_pattern';
+import { ContextApp } from './context_app';
+import { setServices } from '../../../kibana_services';
+import { DiscoverServices } from '../../../build_services';
+import {
+  CONTEXT_TIE_BREAKER_FIELDS_SETTING,
+  SEARCH_FIELDS_FROM_SOURCE,
+} from 'src/plugins/discover/common';
+import { indexPatternsMock } from '../../../__mocks__/index_patterns';
+import { act } from 'react-dom/test-utils';
+
+const mockFilterManager = createFilterManagerMock();
+const mockNavigationPlugin = { ui: { TopNavMenu: mockTopNavMenu } };
+
+describe('ContextApp test', () => {
+  const defaultProps = {
+    indexPattern: indexPatternMock,
+    indexPatternId: 'the-index-pattern-id',
+    anchorId: 'mocked_anchor_id',
+  };
+
+  const topNavProps = {
+    appName: 'context',
+    showSearchBar: true,
+    showQueryBar: false,
+    showFilterBar: true,
+    showSaveQuery: false,
+    showDatePicker: false,
+    indexPatterns: [indexPatternMock],
+    useDefaultBehaviors: true,
+  };
+
+  beforeEach(() => {
+    setServices(({
+      data: {
+        search: {
+          searchSource: {
+            createEmpty: jest.fn(),
+          },
+        },
+      },
+      capabilities: {
+        discover: {
+          save: true,
+        },
+      },
+      indexPatterns: indexPatternsMock,
+      toastNotifications: { addDanger: () => {} },
+      navigation: mockNavigationPlugin,
+      core: { notifications: { toasts: [] } },
+      history: () => {},
+      filterManager: mockFilterManager,
+      uiSettings: {
```

Comment:
We already have a `ui_settings` mock that we could possibly reuse here:
`https://github.com/elastic/kibana/blob/master/src/plugins/discover/public/__mocks__/ui_settings.ts#L12`
by adding those keys in it.